### PR TITLE
Avoid using uncessary `get_values(MODEL_MAPPING)`

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

When we switch to torch 2.2 in daily CI, `natten` causing issue and I temporarily disabled its installation. However, some test files use `model_class in get_values(MODEL_MAPPING)` which will fail with

> RuntimeError: Failed to import transformers.models.dinat.modeling_dinat because of the following error (look up to see its traceback):

even if those models are not requiring `natten`. 

The usage of 

> get_values(MODEL_MAPPING)`

is slow and unnecessary, we can just use the names (string) rather than import them to get the classes.

This PR changes them to use the class name strings.

### CODE REVIEW

**Comment by ydshieh** [comment]
cc @NielsRogge For future vision models, let's avoid using `get_values()` 🙏

**Comment by ydshieh** [comment]
Hmm, sorry there is still a tiny issue. Working on it

File: tests/models/perceiver/test_modeling_perceiver.py [hunk 1]
```diff
@@ -727,11 +731,14 @@ def test_correct_missing_keys(self):
...
-                *get_values(MODEL_FOR_SEQUENCE_CLASSIFICATION_MAPPING),
-                *get_values(MODEL_FOR_IMAGE_CLASSIFICATION_MAPPING),
+            if model_class.__name__ in [
+                "PerceiverForOpticalFlow",
+                "PerceiverForMultimodalAutoencoding",
+                *MODEL_FOR_SEQUENCE_CLASSIFICATION_MAPPING_NAMES.values(),
+                "PerceiverForImageClassificationLearned",
+                "PerceiverForImageClassificationFourier",
+                "PerceiverForImageClassificationConvProcessing",
+                *MODEL_FOR_IMAGE_CLASSIFICATION_MAPPING_NAMES.values(),
```
Comment (ydshieh) [inline tests/models/perceiver/test_modeling_perceiver.py:741]: Using stuff like `MODEL_FOR_IMAGE_CLASSIFICATION_MAPPING_NAMES.values()` is not reliable, as an element in `MODEL_FOR_IMAGE_CLASSIFICATION_MAPPING_NAMES.values()` maybe a list itself. For example, we have

```
        (
            "perceiver",
            (
                "PerceiverForImageClassificationLearned",
                "PerceiverForImageClassificationFourier",
                "PerceiverForImageClassificationConvProcessing",
            ),
        ),
```
in `MODEL_FOR_IMAGE_CLASSIFICATION_MAPPING_NAMES`.


It's probably better to have something like `_get_names_for_values`: similar to `get_values` in `src/transformers/models/auto/auto_factory.py` but only return names without loading the classes.
 
I could do that in a follow up PR, or any better suggestion.

Comment (ArthurZucker) [reply tests/models/perceiver/test_modeling_perceiver.py:741]: yeah it's faster also to not load the classes and just compare strings

**Review by ArthurZucker — APPROVED** [review]
Love that it is simpler 😉

**Pull request merged**